fmf/utils.py

```diff
@@ -202,9 +202,11 @@ def evaluate(expression, data, _node=None):
     Expects data dictionary which will be used to populate local
     namespace. Used to provide flexible conditions for filtering.
     """
-    locals().update(data)
+    # Since python3.13 and https://peps.python.org/pep-0667/
+    _locals = copy.deepcopy(locals())
```

Comment:
I don’t `deepcopy()` is needed. `locals()` should regurn an independent snapshot of local namespace, i.e. we would be making (and updating) a copy of dictionary no other part of code can ever see. Seems superfluous.

Reply:
In other words, I agree we need to store `locals()` somewhere if we need to both update it and pass it to a function call,  ut IIUIC, we don’t need to work with a copy of said dictionary.